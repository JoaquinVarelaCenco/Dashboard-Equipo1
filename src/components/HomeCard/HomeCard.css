.itemRow{
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 50px;
    margin: 10px 5px;
    color: var(--clear-Color);
    background-color: var(--clear-Background);
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    font-size: 13px;
    border-radius: 15px;
    transition: all ease-in-out .5s;
    padding-right: 5px;
}

.itemRow__count{
    display: flex;
    align-items: center;
    gap: 10px;
    margin-left: 6px;
    min-width: 100px;
}

.itemRow__count img{
    width: 25px;
}

.itemRow__buttons{
    display: flex;
justify-content: flex-end;
gap: 10px;
    width: 100%;
    padding-left: 10px;
}

.itemRow__buttons button{
    border: none;
    padding: 8px 0;
    font-size: 8px;
    border-radius: 15px;
    min-width: 80px;
}


@media (min-width: 530px){
    .itemRow{
        flex-direction: row;
        padding: 0 20px;
        font-size: 22px;
        height: 70px;
    }
    
    .itemRow__count img{
        width: 45px;
    }
    
    .itemRow__buttons{
        justify-content: flex-end;
        width: 60%;
    }

    .itemRow__buttons button{
        width: 130px;
        font-size: 14px;
        padding: 12px 10px;
        margin: 0 6px;
    } 
}




/* DarkMode */

.dark .itemRow{
    background-color: #3c3c3c;
    color: var(--dark-Color);
    box-shadow: 0px 2px 2px rgba(117, 117, 117, 0.35);
}

.dark .itemRow__count img{
    filter: invert();
}