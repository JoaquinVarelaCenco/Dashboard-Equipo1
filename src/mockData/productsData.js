module.exports = [
    {
      "id": 2,
      "title": "iPhone X",
      "description": "SIM-Free, Model A19211 6.5-inch Super Retina HD display with OLED technology A12 Bionic chip with ...",
      "price": 899,
      "rating": { "rate": 4.44, "count": 126 },
      "stock": 34,
      "category": "smartphones",
      "images": [],
      "lastModified": "2022-10-12T01:07:11.630Z"
    },
    {
      "id": 3,
      "title": "Samsung Universe 9",
      "description": "Samsung's new variant which goes beyond Galaxy to the Universe",
      "price": 1249,
      "rating": { "rate": 4.09, "count": 845 },
      "stock": 36,
      "category": "smartphones"
    },
    {
      "id": 23,
      "title": "Orange Essence Food Flavou",
      "description": "Specifications of Orange Essence Food Flavour For Cakes and Baking Food Item",
      "price": 14,
      "rating": { "rate": 4.85, "count": 1645 },
      "stock": 26,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/23/1.jpg",
        "https://dummyjson.com/image/i/products/23/2.jpg",
        "https://dummyjson.com/image/i/products/23/3.jpg",
        "https://dummyjson.com/image/i/products/23/4.jpg",
        "https://dummyjson.com/image/i/products/23/thumbnail.jpg"
      ]
    },
    {
      "id": 24,
      "title": "cereals muesli fruit nuts",
      "description": "original fauji cereal muesli 250gm box pack original fauji cereals muesli fruit nuts flakes breakfast cereal break fast faujicereals cerels cerel foji fouji",
      "price": 46,
      "rating": { "rate": 4.94, "count": 11378 },
      "stock": 113,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/24/1.jpg",
        "https://dummyjson.com/image/i/products/24/2.jpg",
        "https://dummyjson.com/image/i/products/24/3.jpg",
        "https://dummyjson.com/image/i/products/24/4.jpg",
        "https://dummyjson.com/image/i/products/24/thumbnail.jpg"
      ],
      "lastModified": "2022-10-17T14:31:36.884Z"
    },
    {
      "id": 25,
      "title": "Gulab Powder 50 Gram",
      "description": "Dry Rose Flower Powder Gulab Powder 50 Gram â€¢ Treats Wounds",
      "price": 70,
      "rating": { "rate": 4.87, "count": 568 },
      "stock": 47,
      "category": "groceries",
      "images": [
        "https://dummyjson.com/image/i/products/25/1.png",
        "https://dummyjson.com/image/i/products/25/2.jpg",
        "https://dummyjson.com/image/i/products/25/3.png",
        "https://dummyjson.com/image/i/products/25/4.jpg",
        "https://dummyjson.com/image/i/products/25/thumbnail.jpg"
      ]
    }
  ]
  