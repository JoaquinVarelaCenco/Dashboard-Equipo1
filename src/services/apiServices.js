const url = "http://localhost:5000/api/";
const getProducts = () => {
  return fetch(`${url}product`).then((res) => res.json())
};

const getProductById = (id) => {
  return fetch(`${url}product/${id}`).then((res) => res.json());
};

const deleteProduct = (productId) => {
  return fetch(`${url}product?id=${productId}`, {
    method: "DELETE",
  }).then((res) => res.json());
};

const updateProduct = (body) => {
  return fetch(`${url}product`, {
    method: "PUT",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  }).then((res) => res.json());
};

const newProduct = (body) => {
  return fetch(`${url}product`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  }).then((res) => res.json());
};

module.exports = {
  getProducts,
  getProductById,
  deleteProduct,
  updateProduct,
  newProduct,
};
