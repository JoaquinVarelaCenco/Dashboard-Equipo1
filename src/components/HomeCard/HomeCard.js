import { Link } from 'react-router-dom'
import Button from '../Button/Button';

import './HomeCard.css'

const HomeCard = (props) => {

    const { count, 
            title, 
            linkBtnList, 
            textBtnAdd, 
            linkBtnAdd, 
            img } = props;

    return (
    <div className='itemRow'>
        <div className='itemRow__count'>
            <img src= {img} alt="Icono de Card"/>
            <p><strong>{count}</strong> {title}</p>
        </div>
        <div className='itemRow__buttons'>
            <Link to={linkBtnList}>
                <Button 
                    title={"Ver listado"}
                />
            </Link>
            <Link to={linkBtnAdd}>
                <Button 
                    title={textBtnAdd}
                />
            </Link>
        </div>
    </div>
    )
}

export default HomeCard