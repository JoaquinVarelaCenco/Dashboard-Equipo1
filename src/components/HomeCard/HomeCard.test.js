import { render, screen } from "@testing-library/react";
import HomeCard from './HomeCard'
import {BrowserRouter} from 'react-router-dom';
import { getProducts } from "../../services/apiServices";
import productsData from '../../mockData/productsData'


//mocks
jest.mock('../../services/apiServices');

let component;

describe("Test formulario de productos", ()=>{

    beforeEach( async()=>{
            await getProducts.mockResolvedValue({ json: () => new Promise(resolve => resolve({ ...productsData }) )})
            
            component = render(<HomeCard
                count={productsData.length}
                title={"Productos"}
                linkBtnList= {"/products"}
                textBtnAdd = {"Agregar Producto"}
                linkBtnAdd = {"/products/new"}
                img = {"as"}
            />, 
        { wrapper: BrowserRouter })

        
    })

    test("El componente debe renderizarse correctamente", ()=>{
        const { container } = component
        expect(container).toMatchSnapshot()
    })

    test("Los links deben mostrarse y dirigir al lugar esperado", ()=>{
        const buttons = screen.getAllByRole('link');
        const btnViewList = buttons[0];
        const btnAddProduct = buttons[1];
        
        expect(btnViewList.getAttribute('href')).toBe('/products');
        expect(btnAddProduct.getAttribute('href')).toBe('/products/new');
    })   

    test("Se debe mostrar la cantidad del item correctamente", async ()=>{

        const total = productsData.length
        const countValue = screen.getByText(total)

        expect(countValue.innerHTML).toBe( (total.toString()))

    })

})