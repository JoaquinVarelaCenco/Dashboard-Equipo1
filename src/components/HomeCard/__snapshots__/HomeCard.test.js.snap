// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test formulario de productos El componente debe renderizarse correctamente 1`] = `
<div>
  <div
    class="itemRow"
  >
    <div
      class="itemRow__count"
    >
      <img
        src="as"
      />
      <p>
        <strong>
          5
        </strong>
         
        Productos
      </p>
    </div>
    <div
      class="itemRow__buttons"
    >
      <a
        href="/products"
      >
        <button>
          Ver listado
        </button>
      </a>
      <a
        href="/products/new"
      >
        <button>
          Agregar Producto
        </button>
      </a>
    </div>
  </div>
</div>
`;
